pandapower/test/loadflow/test_results.py

```diff
@@ -685,6 +685,34 @@ def test_bus_bus_switch(result_test_network, v_tol=1e-6, i_tol=1e-6, s_tol=5e-3,
     assert abs(net.res_bus.vm_pu.at[b3] - v2) < v_tol
     assert abs(net.res_bus.vm_pu.at[b2] == net.res_bus.vm_pu.at[b2])
 
+    for col in ("p_from_mw", "p_to_mw", "q_from_mvar", "q_to_mvar", "i_ka", "loading_percent"):
+        assert col in net.res_switch
+
+    assert isnan(net.res_switch.p_from_mw).all()
+
+    # now test with some switches that have non-zero impedance:
+    net.switch.loc[[2,3], "z_ohm"] = 1e-3
+    pp.runpp(net)
+
+    p_from_ref = -0.29816
+    p_to_ref = 0.29816
+    q_from_ref = -0.96063
+    q_to_ref = 0.96063
+
+    assert isclose(net.res_switch.p_from_mw.at[2], p_from_ref, rtol=1e-3, atol=1e-6)
+    assert isclose(net.res_switch.p_to_mw.at[2], p_to_ref, rtol=1e-3, atol=1e-6)
+    assert isclose(net.res_switch.q_from_mvar.at[2], q_from_ref, rtol=1e-3, atol=1e-6)
+    assert isclose(net.res_switch.q_to_mvar.at[2], q_to_ref, rtol=1e-3, atol=1e-6)
+
+    net.switch.loc[[2], "closed"] = False
+    pp.runpp(net)
+
+    assert isnan(net.res_switch.p_from_mw[2])
+
+    # return to normal test net with zero impedance switches:
```

Comment:
if you are not testing anything after this it is not necessary to have this part